CruCentralCoast/Community/CommunityGroups/CommunityGroupDetailsVC.swift

```diff
@@ -29,10 +29,18 @@ class CommunityGroupDetailsVC: UIViewController, MFMessageComposeViewControllerD
     }
     
     @IBAction func didTapContactLeader() {
+        let userIsLoggedIn = LoginManager.instance.user != nil
+        
+        if !userIsLoggedIn {
+            print("got here")
+            LoginManager.instance.presentLogin(from: self)
```

Comment:
You will want to return after this statement to prevent the rest of the function from executing

Reply:
what do you think the flow should be if a user is signing in for the first time when tapping this button?

because right now they push the button, sign in, modal goes away, then they have to push the button a second time to be taken into messages

Reply:
@tydahlwave

Reply:
That's fine for now. Ideally, upon login, the action should complete and they should be taken to the contact form

Reply:
okay cool, I agree